homeassistant/components/modbus/climate.py

```diff
@@ -387,3 +399,11 @@ async def _async_read_register(
             return None
         self._attr_available = True
         return float(self._value)
+
+    def _GetMeTargetTempRegister(self) -> int:
+        """Get the target temp register for a specific HVAC mode."""
+        return int(
```

Comment:
It is prepared in the validator so the int() should not be needed.

In general the function should be dissolved and the array indexing made directly,

Reply:
Hi @janiversen, this morning I waked up having the idea to restore the try/except block, but I see that you prefer to remove completely the routine. Are you confident that we can delete the whole defensive approach, I mean the try/except and the cast?

Reply:
you should know that it's your PR. But you are defending against yourself?